lib/cli-engine/cli-engine.js

```diff
@@ -56,6 +56,8 @@ const validFixTypes = new Set(["directive", "problem", "suggestion", "layout"]);
 /** @typedef {import("../shared/types").Plugin} Plugin */
 /** @typedef {import("../shared/types").RuleConf} RuleConf */
 /** @typedef {import("../shared/types").Rule} Rule */
+/** @typedef {import("../shared/types").LintResult} LintResult */
```

Comment:
Thanks @mdjermanovic, I missed that one! The definition of `LintResult` in cli-engine.js is identical to the shared one, except that it does not include `usedDeprecatedRules`. This makes sense somehow, since that property is attached later outside the CLIEngine:

https://github.com/eslint/eslint/blob/5d60812d440762dff72420714273c714c4c5d074/lib/eslint/eslint.js#L408-L410

The other definition could be better modeled with type inheritance, but changing it would be outside the scope of this PR, I think. I will remove the import if nobody comes up with a different suggestion.